### PR TITLE
Create Ethnologie française (French)

### PR DESCRIPTION
**Pull request by VRD75**
To csl-bot:
I met one mistake by the validation process at line 44. However, the script works in practice.

### CODE REVIEW

**Comment by rmzelle** [comment]
Thanks for contributing!

Could you change the style ID and "self" link to use "ethnologie-francaise" instead of "ethnologiefrancaise"? And could you change the title from "Ethnologie Française (French)" to "Ethnologie française (French)". Finally, could you change the file name from "Ethnologie française (French)" to "ethnologie-francaise.csl"? (see also https://help.github.com/articles/renaming-a-file/)

**Comment by VRD75** [comment]
To rmzelle:
I encountered several problems to submit my style. The process is too time-consuming. My original file is named etnologiefrancaise.csl but I wish it to appear in the Zotero repository "Ethnology Française (French)".

**Comment by rmzelle** [comment]
@VRD75, sorry, we try to make it as easy as possible to submit styles. Is there anything in particular you find time-consuming?

If you click the pencil/edit icon at https://github.com/VRD75/styles/blob/patch-1/Ethnologie%20fran%C3%A7aise%20(French), you should be able to quickly correct the three issues I highlighted above.

**Comment by VRD75** [comment]
To rmzelle : 
to understand how sumbit a style only, it's time-consuming.

**Comment by VRD75** [comment]
To csl-bot:
I met one mistake by the validation process at line 44. However, the script works in practice.
 
De : csl-bot [mailto:notifications@github.com] 
Envoyé : mardi 1 décembre 2015 18:52
À : citation-style-language/styles
Cc : VRD75
Objet : Re: [styles] Create Ethnologie française (French) (#1788)
 
Oops! Travis CI (our automated test system) has found some issues with your submission. Please check the Travis test report <https://travis-ci.org/citation-style-language/styles/builds/94215132>  and make sure your styles validate <https://github.com/citation-style-language/styles/wiki/Validation>  and follow all our other Style Requirements <https://github.com/citation-style-language/styles/wiki/Style-Requirements> . Please ask for help if there is anything in the test report you don't understand.
When you're ready to make changes, you can easily update this pull request by visiting the "Files changed" tab above. Then click the "View" button of the file you wish to change, and click on the pencil icon (see below) that appears in the next screen to start editing.
 <https://cloud.githubusercontent.com/assets/77951/10228312/da729b6a-683f-11e5-8219-748562173ae8.png> image

**Comment by VRD75** [comment]
I hope that I have correctly answered at your requests for renaming. Sorry for my bad english.
 
De : Rintze M. Zelle [mailto:notifications@github.com] 
Envoyé : mardi 1 décembre 2015 19:05
À : citation-style-language/styles
Cc : VRD75
Objet : Re: [styles] Create Ethnologie française (French) (#1788)
 
@VRD75 <https://github.com/VRD75> , sorry, we try to make it as easy as possible to submit styles. Is there anything in particular you find time-consuming?
If you click the pencil/edit icon at https://github.com/VRD75/styles/blob/patch-1/Ethnologie%20fran%C3%A7aise%20(French), you should be able to quickly correct the three issues I highlighted above.

File: Ethnologie française (French) [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  </locale>
+  <macro name="author">
+    <names variable="author" delimiter=", ">
+      <name and="text" name-as-sort-order="all" sort-separator=" " delimiter=", ">		
+		<name-part name="family" font-variant="small-caps" />
+		<name-part name="given" />
+		</name>
+      <substitute>
+        <names variable="editor"/>
+		<label form="short" text-case="lowercase" prefix=" (" suffix=")"/>
```
Comment (adam3smith) [inline Ethnologie française (French)]: you'll want to take this line and move it up after line 38. You can't have a label in substitute.

**Comment by adam3smith** [comment]
We do need the style to validate before we can accept it, see my not above. Remember, these go out to hundreds of thousands of users of more than 20 different software products. We do need to insist on pretty high standards for contributions if that's going to work.

**Comment by VRD75** [comment]
Franchement, cela devient trop compliqué et chronophage (time-consuming). L’erreur signalée dans la procédure de validation ligne 44 n’existe pas en réalité puisque cela fonctionne dans la pratique.
Si cela ne convient pas, j’abandonne ma soumission qui n’était faite que pour aider des collègues gratuitement. Mais finalement, tant pis, je garderai cela pour moi, car cela fait plusieurs heures cumulées que j’essaie de soumettre mon script.
Merci quand même de votre aide.
VRD
 
De : csl-bot [mailto:notifications@github.com] 
Envoyé : mardi 1 décembre 2015 19:18
À : citation-style-language/styles
Cc : VRD75
Objet : Re: [styles] Create Ethnologie française (French) (#1788)
 
Oops! Travis CI (our automated test system) has found some issues with your submission. Please check the Travis test report <https://travis-ci.org/citation-style-language/styles/builds/94220520>  and make sure your styles validate <https://github.com/citation-style-language/styles/wiki/Validation>  and follow all our other Style Requirements <https://github.com/citation-style-language/styles/wiki/Style-Requirements> . Please ask for help if there is anything in the test report you don't understand.
When you're ready to make changes, you can easily update this pull request by visiting the "Files changed" tab above. Then click the "View" button of the file you wish to change, and click on the pencil icon (see below) that appears in the next screen to start editing.
 <https://cloud.githubusercontent.com/assets/77951/10228312/da729b6a-683f-11e5-8219-748562173ae8.png> image

**Comment by adam3smith** [comment]
sorry about that. But while we try to make this as clear and simple as possible, we can't lower standards for submissions.

**Comment by VRD75** [comment]
Yes, I know and I followed all these instructions.
 
 
De : Rintze M. Zelle [mailto:notifications@github.com] 
Envoyé : mardi 1 décembre 2015 19:35
À : citation-style-language/styles
Cc : VRD75
Objet : Re: [styles] Create Ethnologie française (French) (#1788)
 
@VRD75 <https://github.com/VRD75> , you did find our step-by-step instructions at https://github.com/citation-style-language/styles/blob/master/CONTRIBUTING.md?

**Comment by rmzelle** [comment]
As highlighted by @adam3smith and me above, there are still two issues:

* the style file name should be "ethnologie-francaise.csl" (instructions on renaming files can be found at https://help.github.com/articles/renaming-a-file/)
* to validate, the style should use

```
<macro name="author">
  <names variable="author" delimiter=", ">
    <name and="text" name-as-sort-order="all" sort-separator=" " delimiter=", ">
      <name-part name="family" font-variant="small-caps"/>
      <name-part name="given"/>
    </name>
    <label form="short" text-case="lowercase" prefix=" (" suffix=")"/>
    <substitute>
      <names variable="editor"/>
    </substitute>
  </names>
</macro>
```

instead of

```
<macro name="author">
  <names variable="author" delimiter=", ">
    <name and="text" name-as-sort-order="all" sort-separator=" " delimiter=", ">
      <name-part name="family" font-variant="small-caps"/>
      <name-part name="given"/>
    </name>
    <substitute>
      <names variable="editor"/>
      <label form="short" text-case="lowercase" prefix=" (" suffix=")"/>
    </substitute>
  </names>
</macro>
```

Since you already made some successful additional commits to this pull request, it really shouldn't take you very long to make these two extra changes. After that the style should be good to go. Let us know if you have trouble with any of these requests. We're always happy to provide guidance.

(we ask people who contribute styles to correct any issues themselves, because it makes it clearer who contributed each style. @adam3smith and I also maintain this repository in our spare time, and we can only keep doing this if contributors do most of the work, since we get hundreds of style contributions each year. Some contributors also enjoy learning the process, especially since most of the steps are similar for other open source projects)

**Comment by rmzelle** [comment]
(and csl-bot is an automated system, not a person)

**Comment by VRD75** [comment]
Thank you all for your advice and your patience.
I corrected my file.
Best regards,
VRD
 
De : Rintze M. Zelle [mailto:notifications@github.com] 
Envoyé : mardi 1 décembre 2015 20:40
À : citation-style-language/styles
Cc : VRD75
Objet : Re: [styles] Create Ethnologie française (French) (#1788)
 
(and csl-bot is an automated system, not a person)

**Comment by rmzelle** [comment]
Can you also change the style's file name? When you're editing the style you can do this in the field just above the style itself: https://help.github.com/assets/images/help/repository/changing-file-name.png

**Comment by VRD75** [comment]
I just changed the style’s file name.
Thanks for your help.
VRD
 
De : Rintze M. Zelle [mailto:notifications@github.com] 
Envoyé : mercredi 2 décembre 2015 15:21
À : citation-style-language/styles
Cc : VRD75
Objet : Re: [styles] Create Ethnologie française (French) (#1788)
 
Can you also change the style's file name? When you're editing the style you can do this in the field just above the style itself: https://help.github.com/assets/images/help/repository/changing-file-name.png

**Comment by rmzelle** [comment]
Great. One last request (I promise): can you change `<link href="http://www.zotero.org/styles/ethnologiefrancaise" rel="self"/>` into `<link href="http://www.zotero.org/styles/ethnologie-francaise" rel="self"/>`? (use the edit/pencil icon at https://github.com/VRD75/styles/blob/patch-1/ethnologie-francaise.csl )

**Comment by rmzelle** [comment]
Oh, sorry, I missed one more error. The license line should be:

`<rights license="http://creativecommons.org/licenses/by-sa/3.0/">This work is licensed under a Creative Commons Attribution-ShareAlike 3.0 License</rights>`

(this must be on a single line)

not

`<rights>This work is licensed under a Creative Commons Attribution-Share Alike 3.0 License: http://creativecommons.org/licenses/by-sa/3.0/</rights>`

(I should have paid more attention when looking at the previous Travis error report)

**Comment by rmzelle** [comment]
(and the reason these last few errors only pop up now is that the validation tool only checks file with a ".csl" extension in detail, so that had to be corrected first)

**Comment by rmzelle** [comment]
(it should read "ShareAlike", not "Share Alike")

**Comment by rmzelle** [comment]
Yay!!!

Sorry about that, usually we go fewer rounds of this back and forth.

**Comment by VRD75** [comment]
Merci infiniment pour tout. J’ai archive l’ensemble de vos remarques pour vous éviter toutes ces démarches la prochaine fois.
Thank you so much for your help.
 
 
De : Rintze M. Zelle [mailto:notifications@github.com] 
Envoyé : mercredi 2 décembre 2015 21:43
À : citation-style-language/styles
Cc : VRD75
Objet : Re: [styles] Create Ethnologie française (French) (#1788)
 
Yay!!!
Sorry about that, usually we go fewer rounds of this back and forth.

**Pull request merged**